app/services/scheduler_service.rb

```diff
@@ -0,0 +1,154 @@
+class SchedulerService
+  require "google/apis/calendar_v3"
+  require "googleauth"
+  require "googleauth/stores/file_token_store"
+  require "date"
+  require "fileutils"
+
+  attr_reader :current_user, :weekdays, :hour_scheduled_times
+
+  def initialize(current_user)
+    @current_user = current_user
+    @weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+    @hour_scheduled_times = [("08:01".."08:59"), ("09:01".."09:59"),("10:01".."10:59"),("11:01".."11:59"),("12:01".."12:59"),( "13:01".."13:59"),( "14:01".."14:59"), ("15:01".."15:59"), ("16:01".."16:59"), ("17:01".."17.59"), ("18:01".."18:59"),( "19:01".."19:59"), ("20:01".."20:59")]
+  end
+
+  # sets up initial Google Calendar Service with current user's google token
+  def get_calendar_service
+    service ||= Google::Apis::CalendarV3::CalendarService.new
+    service.client_options.application_name = "Treat Yo Self"
+    service.authorization = @current_user.google_token
+    service
+  end
+
+  # Uses Google Service to find a list of user events within current week time period
+  def find_user_events
+    calendar_id = "primary"
+    response = get_calendar_service.list_events( calendar_id,
+                                   time_min: DateTime.now.rfc3339,
+                                   time_max: (DateTime.now + 1.week).rfc3339,
+                                   single_events: true,
+                                   order_by: "startTime" )
+  end
+
+  # Sets available times from 8 a.m. - 8 p.m. as a string and puts that
+  # in an array by the minute i.e. ["08:00", "08:01", "08:02"]
+  def available_time
+    available_times = ("08:00".."20:00").to_a
+    available_times.delete_if do |time|
+    # since this is a string this deletes any minute over 00:59 ex. (09:60 would be deleted)
+    # this returns an array with more realistic time frames
+     time.split("")[3].to_i > 5
+    end
+  end
+
+  # Uses the array of events to subract event time frames from a user's availability.
+  def availability
+    find_user_events.items.reduce({}) do |acc, event|
```

Comment:
Nice use of reduce enum here. Dries up some of the potentially long lines of logic.